pkg/handler/processor/guesser/type_dsse.go

```diff
@@ -0,0 +1,35 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package guesser
+
+import (
+	"encoding/json"
+
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/secure-systems-lab/go-securesystemslib/dsse"
+)
+
+type dsseTypeGuesser struct{}
+
+func (_ *dsseTypeGuesser) GuessDocumentType(blob []byte, format processor.FormatType) processor.DocumentType {
+	var envelope dsse.Envelope
+	if json.Unmarshal(blob, &envelope) == nil && format == processor.FormatJSON {
+		if envelope.Payload != "" && envelope.PayloadType != "" {
```

Comment:
We should also check that there is atleast one signature for a valid payload

Reply:
yea.. i don't like how there isn't really a good way to guess that it's a DSSE based on the fields (they are all so generic). Yea I think its fair to just check that there's at least 1 signature. @SantiagoTorres i'm guessing this is for space reasons? but may be weird without versioning if there's a version 2.0?

(a minor nit here is that it doesn't actually have to be a valid envelope, but a side effect of it could be - since schema validation is handled again by the processor).